app/vacancy_sources/vacancy_source/source/every.rb

```diff
@@ -79,12 +79,20 @@ def main_organisation(item)
     schools_for(item).one? ? schools_for(item).first : multi_academy_trust(item)
   end
 
-  def job_role(item)
-    item["jobRole"].presence
-      &.gsub("headteacher", "senior_leader")
-      &.gsub("head_of_year", "middle_leader")
-      &.gsub("learning_support", "education_support")
-      &.gsub(/\s+/, "")
+  def job_roles_for(item)
+    role = item["jobRole"]&.strip
+    return [] if role.blank?
+
+    # Translate legacy senior/middle leader into all the granular roles split from them
+    return Vacancy::SENIOR_LEADER_JOB_ROLES if role.include? "senior_leader"
+    return Vacancy::MIDDLE_LEADER_JOB_ROLES if role.include? "middle_leader"
+
+    Array.wrap(role.gsub("deputy_headteacher_principal", "deputy_headteacher")
+                   .gsub("assistant_headteacher_principal", "assistant_headteacher")
+                   .gsub("headteacher_principal", "headteacher")
+                   .gsub(/head_of_year_or_phase|head_of_year/, "head_of_year_or_phase")
```

Comment:
[Minor] I think `head_of_year_or_phase` is redundant here

Reply:
This is tricky to see in the code and easier to understand with a breaking test (that is how I noticed :sweat: ):
When you have `"head_of_year_or_phase".gsub(/head_of_year/, "head_of_year_or_phase")` the output is `head_of_year_or_phase_or_phase`, because it replaces the `head_of_year` substring.

With this regexp `/head_of_year_or_phase|head_of_year/`, the `head_of_year` matching is a fallback, so it doesn't apply if the left side (`head_of_year_or_phase`) matches, avoiding the issue explained above.

We could also achieve this by improving the matcher, to be fair. Maybe we could clean the spaces as a first step (instead of the current last step) and then do a `.gsub(/^head_of_year$/,...)` regexp to be more precise about not replacing it if it is a substring?

Reply:
Ahhhhh, I see, that makes sense! I don't think we need to add anymore complicated regex, I just didn't think of this scenario 😅